src/sync.d

```diff
@@ -3375,9 +3679,31 @@ final class SyncEngine
 		uploadFailed = false;
 	
 		Item parent;
-		// Check the database for the parent
-		//enforce(itemdb.selectByPath(dirName(path), defaultDriveId, parent), "The parent item is not in the local database");
-		if ((dryRun) || (itemdb.selectByPath(dirName(path), defaultDriveId, parent))) {
+		bool parentPathFoundInDB = false;
+		// Check the database for the parent path
+		// What parent path to use?
+		string parentPath = dirName(path);		// will be either . or something else
+		if (parentPath == "."){
+			// Assume this is a new file in the users configured sync_dir root
+			// Use client defaults
+			parent.id = defaultRootId;  // Should give something like 12345ABCDE1234A1!101
+			parent.driveId = defaultDriveId;  // Should give something like 12345abcde1234a1
+			parentPathFoundInDB = true;
+		} else {
+			// Query the database using each of the driveId's we are using
+			foreach (driveId; driveIDsArray) {
+				// Query the database for this parent path using each driveId
+				Item dbResponse;
+				if(itemdb.selectByPathWithRemote(parentPath, driveId, dbResponse)){
+					// parent path was found in the database
+					parent = dbResponse;
+					parentPathFoundInDB = true;
+				}
+			}
+		}
+				
```

Comment:
The very same code appears in line 3481, and maybe other places, can we get rid of these duplications? (Not necessary for this PR, but could be a separate ;-) )